### PR TITLE
chore(deps): update vite and rollup to latest

### PR DESCRIPTION
**Pull request by huijiewei**
I update vite, rollup and other deps to latest.

test has spa error. because test vite config file use adapter: process.env.START_ADAPTER, i remove this option. test is ok.

### CODE REVIEW

**Comment by huijiewei** [comment]
#660

**Comment by birkskyum** [comment]
Can you elaborate a bit on what you did to make the spa test work? Where in the code

**Comment by birkskyum** [comment]
For me this actually passes even with the spa-test left as is.

File: packages/start/bin.cjs [hunk 1]
```diff
@@ -519,7 +519,7 @@ async function resolveConfig({ configFile, root, mode, command }) {
     DEBUG('config file: "%s"', configFile);
   }
 
-  let config = await vite.resolveConfig({ mode, configFile, root }, command);
```
Comment (birkskyum) [inline packages/start/bin.cjs:522]: This line, passing the mode to ViteJS, is what fixes the spa-test. It doesn't seem to need the removal of the adapter.

File: packages/start/bin.cjs [hunk 2]
```diff
@@ -519,7 +519,7 @@ async function resolveConfig({ configFile, root, mode, command }) {
...
+  let config = await vite.resolveConfig({ mode, configFile, root }, command, mode, mode);
```
Comment (birkskyum) [inline packages/start/bin.cjs:522]: This line, passing the mode to ViteJS, is what fixes the spa-test. It doesn't seem to need the removal of the adapter.

**Comment by birkskyum** [comment]
I've made a reduced PR here #767 , that only handles Vite 4. I was a bit worried that bumping this many dependencies at a time could introduce issues, like node-resolve 15: https://github.com/solidjs/solid-start/commit/20de367addbba2fc3876572bcb746ec8efee77e2

**Comment by rtritto** [comment]
Fix #713

**Comment by birkskyum** [comment]
This PR is a bit too big and has also fallen behind the main branch. @rtritto , if you make a new smaller PR that only has this small change you seek to fix 713, it would be a lot easier to review and merge it.

**Comment by ryansolid** [comment]
Yeah I appreciate this but I went through and did a bit myself. It seems like node-resolve may be fine now. We are going to try it. I've updated everything else.

**Pull request closed without merging**